File: client-android/app/src/main/java/stasis/client_android/activities/views/context/EntryAction.kt
package stasis.client_android.activities.views.context

import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes

data class EntryAction(
    @DrawableRes val icon: Int,
    val name: String,
    val description: String,
    val handler: () -> Unit,
    @ColorRes val color: Int?
) {
    companion object {
        operator fun invoke(
            @DrawableRes icon: Int,
            name: String,
            description: String,
            handler: () -> Unit,
        ): EntryAction = EntryAction(
            icon = icon,
            name = name,
            description = description,
            handler = handler,
            color = null
        )
    }
}

File: client-ui/lib/pages/components/context/entry_action.dart
import 'package:flutter/cupertino.dart';

class EntryAction {
  const EntryAction({
    required this.icon,
    required this.name,
    required this.description,
    required this.handler,
    this.color,
  });

  final IconData icon;
  final String name;
  final String description;
  final void Function() handler;
  final Color? color;
}
